warehouse/packaging/models.py

```diff
@@ -77,6 +77,38 @@ class Role(db.Model):
     project = orm.relationship("Project", lazy=False)
 
 
+class RoleInvitationStatus(enum.Enum):
+
+    Pending = "pending"
+    Expired = "expired"
+
+
+class RoleInvitation(db.Model):
+
+    __tablename__ = "role_invitations"
+    __table_args__ = (
+        Index("role_invitations_user_id_idx", "user_id"),
+        UniqueConstraint(
+            "user_id", "project_id", name="_role_invitations_user_project_uc"
+        ),
+    )
+
+    __repr__ = make_repr("invite_status", "user", "project")
+
+    invite_status = Column(Text, nullable=False)
```

Comment:
This should probably be an `Enum` instead, e.g: https://github.com/pypa/warehouse/blob/91a1767337b5e3a0ef01817208aba2b6a4c8f554/warehouse/packaging/models.py#L486-L497